modules/publisher/views.py

```diff
@@ -12,44 +13,50 @@
     ApiResponseDecodeError,
     ApiResponseError,
     ApiResponseErrorList,
+    MacaroonNeedsRefresh
 )
 
 
 def get_account():
-    account = api.get_account(flask.session)
-    if 'redirect' in account:
-        return account['redirect']
-
-    error_list = []
-    if 'error_list' in account:
-        for error in account['error_list']:
-            if error['code'] == 'user-not-ready':
-                if 'has not signed agreement' in error['message']:
-                    return flask.redirect('/account/agreement')
-                elif 'missing namespace' in error['message']:
-                    return flask.redirect('/account/username')
-            else:
-                error_list.append(error)
-
-    user_snaps = []
-    if '16' in account['snaps']:
-        user_snaps = account['snaps']['16']
-
-    flask_user = flask.session['openid']
+    try:
+        account = api.get_account(flask.session)
```

Comment:
Does the whole function need to be in a try block or can it be limited to a smaller section, maybe this line?

Reply:
For now, yes because this part is not totally refactored yet